src/gbm/gbtree_model.cc

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright 2019 by Contributors
+ */
+#include "xgboost/json.h"
+#include "xgboost/logging.h"
+#include "gbtree_model.h"
+
+namespace xgboost {
+namespace gbm {
+
+void GBTreeModel::SaveModel(dmlc::Stream* fo) const {
+  CHECK_EQ(param.num_trees, static_cast<int>(trees.size()));
+  fo->Write(&param, sizeof(param));
+  for (const auto & tree : trees) {
+    tree->SaveModel(fo);
+  }
+  if (tree_info.size() != 0) {
+    fo->Write(dmlc::BeginPtr(tree_info), sizeof(int) * tree_info.size());
+  }
+}
+
+void GBTreeModel::LoadModel(dmlc::Stream* fi) {
+  CHECK_EQ(fi->Read(&param, sizeof(param)), sizeof(param))
+      << "GBTree: invalid model file";
+  parameter_initialised_ = true;
+  trees.clear();
+  trees_to_update.clear();
+  for (int i = 0; i < param.num_trees; ++i) {
+    std::unique_ptr<RegTree> ptr(new RegTree());
+    ptr->LoadModel(fi);
+    trees.push_back(std::move(ptr));
+  }
+  tree_info.resize(param.num_trees);
+  if (param.num_trees != 0) {
+    CHECK_EQ(
+        fi->Read(dmlc::BeginPtr(tree_info), sizeof(int) * param.num_trees),
```

Comment:
Note that if we would write on machine with sizeof(int) == 8 and read on machine with sizeof(int) == 4 this may fail

Reply:
Will fix it.  This portion of code is legacy and copied from header to avoid "recompiling the world".  Thanks for the suggestion.